packages/suite/src/actions/suite/suiteActions.ts

```diff
@@ -630,11 +638,11 @@ export const authorizeDevice =
         if (response.success) {
             const { state } = response.payload;
             const s = state.split(':')[0];
-            const duplicate = getState().devices.find(
+            const duplicate = devices.find(
```

Comment:
You should select devices here and not before `await TrezorConnect.getDeviceState`. Not sure if it makes a difference here, but better not to have to think about it...